src/Peachpie.Library.Network/CURLConstants.cs

```diff
@@ -1067,11 +1086,58 @@ public override void Apply(Context ctx, HttpWebRequest request)
 
         private static void LoadCookieFile(HttpWebRequest request, PhpStream stream)
         {
-            // TODO: load netscape-like or header-style cookies from stream
+            // load netscape-like or header-style cookies from stream
 
-            // request.CookieContainer.Add() // add parsed cookie
+            // TODO: add support for header-style cookies (`Set-Cookie: ...`).
+            //       currently, only curl's netscape-like cookies are supported.
 
             // request.CookieContainer.SetCookies( ... ) // set header style cookies
+
+            using (var reader = new StreamReader(stream.RawStream, Encoding.ASCII))
+            {
+				string line = String.Empty;
+
+				// iterate over the cookie jar lines
+				while ((line = reader.ReadLine()) != null)
+				{
+
+					// gather current cookie information in a new cookie object
+					var cookie = new Cookie();
+
+					// check for HttpOnly cookies and remove the #HttpOnly prefix
+					if (line.Length < 10 || line.Substring(0, 10) != "#HttpOnly_")
+                        cookie.HttpOnly = false;
+
+					else
+                    {
+						line = line.Substring(10);
+					    cookie.HttpOnly = true;
+					}
+
+					// we only look for non-comment, non-blank, valid cookies per lines
+					if (!String.IsNullOrWhiteSpace(line) && !line.StartsWith("#") && line.Count(c => c == '\t') == 6)
+					{
+						// get tokens in an array and trim them
+						string[] tokens = line.Split('\t').Select(token => token.Trim()).ToArray();
+
+						cookie.Domain = tokens[0]; // The domain that created AND can read the variable.
+						// var subdomainAccess = bool.Parse(tokens[1]); // boolean value indicating if all machines within a given domain can access the variable.
+						cookie.Path = tokens[2] ?? "/"; // The path within the domain that the variable is valid for.
+						cookie.Secure = bool.Parse(tokens[3]); // boolean value indicating if a secure connection with the domain is needed to access the variable.
+
+						// convert expiration date from unix timestamp
+						cookie.Expires = DateTimeOffset.FromUnixTimeSeconds(long.Parse(tokens[4])).UtcDateTime; // TODO: move to DateTimeUtils class?
+
+						// decode cookie name and value
+						cookie.Name = tokens[5];
+						cookie.Value = tokens[6];  // TODO: should HttpUtility.UrlDecode be used here? more research needed
+
+						// add the parsed cookie
+						request.CookieContainer.Add(cookie);
```

Comment:
@jakubmisek it seems that this is not working.  Can you please verify/debug this?

Here's the test case:
[**📦 testcase.zip**](https://github.com/peachpiecompiler/peachpie/files/4118394/testcase.zip)

Reply:
sorry for the delay, just noticed the comment

Reply:
@jakubmisek It's alright. Also I believe I should use a `CookieCollection` first

Reply:
`cookie.Expires` is set to `0` which is in Unix world January 1970. 
Hence `cookie.Expired` is true.
Expired cookie is not added to the container but removed instead:

internally:
```c#
			if (cookie.Expired)
			{
				lock (cookieCollection)
				{
					int num = cookieCollection.IndexOf(cookie);
					if (num != -1)
					{
						cookieCollection.RemoveAt(num);
						m_count--;
					}
				}
			}
```

Reply:
We can change `CookiaContainer` to `CookieCollection`. Just note, eventually it will be copied into the CookieContainer anyways (and internally it uses CookieCollection as well)

Reply:
@jakubmisek I see!

libcurl writes `0` for expired when the `Expires` in [Set-Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Syntax) directive is omitted, e.g. `Set-Cookie: foo=bar; Path=/`.

Thanks for your help; I'll see how this is implemented in libcurl to reflect it here.

Reply:
I see, didn't know that! would be interesting to see how to setup the cookie in .NET to work the same

Reply:
@jakubmisek  I did a little bit of digging around, and it seems it's similar in .NET's [Cookie.Expires Property Remarks](https://docs.microsoft.com/en-us/dotnet/api/system.net.cookie.expires#remarks) reference:

> Setting the `Expires` property to [DateTime.MinValue](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.minvalue) **makes this a session cookie, which is its _default_ value**.

Since this is the default value for the `Expires` method, I quickly verified that it worked by commenting out the offending line.

So, I used something like this to solve the issue:
```cs
long ts = long.Parse(tokens[4]);

cookie.Expires = ts == 0 ? DateTime.MinValue : DateTimeOffset.FromUnixTimeSeconds(ts).UtcDateTime;

// later moved to DateTimeUtils, so:
// cookie.Expires = DateTimeUtils.UnixTimeStampToUtc(ts);
```

BTW, something like this is used the other way around when generating the netscape-style cookies:

```cs
long expires = (c.Expires.Ticks == 0) ? 0 : DateTimeUtils.UtcToUnixTimeStamp(c.Expires);
```